### DISCUSSION TITLE
[Suggestion] Differentiate single product vs product with variants

### DISCUSSION THREAD

**Discussion by sebszocinski** [statamic-rad-pack/shopify#35] [Feature Requests]
I know Shopify handles single products as products with one variant, but there is no way to differentiate an imported product that is single vs one that has variants (ie even just one)

In our case we are displaying information from the variant in our content / description such as it's title and at the moment the only way to tell if a product is single is by the variant title being set to "Default Title" which I don't know is a great idea either? this should probably just be the product title again? or null?

Anyway my suggestion is to add something like `has_variants: false` to the product yaml.

Thoughts?

**Comment by jackabox** [comment]
I've converted this to a discussion, as this fits more in that category than an issue.

If the variant has one only option, this would be a single product. Shopify sets this to **Default Title** and is handled on their side via the import.

A variant is still required to get all the price and stock information, so every product has variants.

**Comment by ryanmitchell** [comment]
This can be solved in antlers by `{{ my_product.variants | count }}` and in Statamic queries by `$query->whereJsonLength('variants', '>', 1)`
